### PR TITLE
Create a dataflow instance specifically for the Serializability library

### PR DESCRIPTION
**Pull request by smowton**
Otherwise because this dataflow instance populates AdditionalTaintStep there is an ever-present danger that a user will stumble into creating a recursive configuration, or at least that by using DataFlow5::Configuration for any other purpose they will needlessly recalculate the Serializability dataflow results.

### CODE REVIEW

**Comment by aschackmull** [comment]
Is number 5 used for anything else at the moment? Otherwise we could consider just renaming it (and possibly renaming number 6 as that is also for a reserved use case IIRC). We have plenty of copies already, so might as well try to keep the number down.

**Comment by smowton** [comment]
They do all have users, though few in number.

**Comment by aschackmull** [comment]
LGTM, but needs resync.

File: java/ql/src/semmle/code/java/frameworks/google/GoogleHttpClientApi.qll [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import java
 import semmle.code.java.Serializability
 import semmle.code.java.dataflow.DataFlow
-import semmle.code.java.dataflow.DataFlow5
+import semmle.code.java.dataflow.internal.DataFlowForSerializability
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/google/GoogleHttpClientApi.qll:4]:
```suggestion
private import semmle.code.java.dataflow.internal.DataFlowForSerializability
```

File: java/ql/src/semmle/code/java/frameworks/jackson/JacksonSerializability.qll [hunk 2]
```diff
@@ -7,7 +7,7 @@ import java
 import semmle.code.java.Serializability
 import semmle.code.java.Reflection
 import semmle.code.java.dataflow.DataFlow
-import semmle.code.java.dataflow.DataFlow5
+import semmle.code.java.dataflow.internal.DataFlowForSerializability
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/jackson/JacksonSerializability.qll:10]:
```suggestion
private import semmle.code.java.dataflow.internal.DataFlowForSerializability
```

**Comment by smowton** [comment]
Accepted suggestions and rebased

**Comment by aschackmull** [comment]
CI still looks very red for some reason.

**Comment by smowton** [comment]
Looks like there were concurrent changes to `DataFlowImpl.qll`. Resynced.

**Review by aschackmull — APPROVED** [review]

**Pull request merged**